### PR TITLE
Add FXIOS-7757 [v122] Add telemetry for adsExposure

### PR DESCRIPTION
**Pull request by PARAIPAN9**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-7757)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/17306)

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [x] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

File: Client/Frontend/Fakespot/FakespotViewModel.swift [hunk 1]
```diff
@@ -186,6 +186,10 @@ class FakespotViewModel {
         return prefs.boolForKey(PrefsKeys.Shopping2023OptIn) ?? false
     }
 
+    private var areAdsEnabled: Bool {
+        settingsCardViewModel.areAdsEnabled
```
Comment (thatswinnie) [inline Client/Frontend/Fakespot/FakespotViewModel.swift:190]: We should check the pref directly here.

File: Client/metrics.yaml [hunk 2]
```diff
@@ -670,6 +670,24 @@ shopping:
       - fx-ios-data-stewards@mozilla.com
     expires: "2024-06-01"
     unit: counter of product pages
+
+  ads_exposure:
+    type: event
+    description: |
+      On a supported product page, the review checker showed analysis, and
+      the ads exposure pref was enabled, or review checker ads were enabled,
```
Comment (thatswinnie) [inline Client/metrics.yaml:678]: We only have one pref so this comment should be updated to reflect that.

File: Client/metrics.yaml [hunk 3]
```diff
@@ -670,6 +670,24 @@ shopping:
...
+      and when we tried to fetch an ad from the ad server, an ad was available.
+      Does not indicate whether the ad was actually shown.
+    bugs:
+      - https://github.com/mozilla-mobile/firefox-ios/issues/17306
+    data_reviews:
+      - https://github.com/mozilla-mobile/firefox-ios/pull/17493
+    notification_emails:
+      - fx-ios-data-stewards@mozilla.com
+    expires: "2024-11-27"
```
Comment (thatswinnie) [inline Client/metrics.yaml:687]:
```suggestion
    expires: "2024-06-01"
```

File: Client/metrics.yaml (hunk 2 above)
Comment (thatswinnie) [reply Client/metrics.yaml:678]: Actually, I think with `review checker ads were enabled` you mean that the Nimbus feature is enabled, correct? In that case ignore my previous comment.

Comment (PARAIPAN9) [reply Client/metrics.yaml:678]: Yes, it's the same description as Desktop.

**Review by thatswinnie — APPROVED** [review]

**Comment by PARAIPAN9** [comment]
# Request for data collection review form

1) What questions will you answer with this data?
- Background: For users who have opted-in to the shopping experiment, when the shopping.experience2023.ads.exposure pref is set to true (it is currently set to false, because the ads server can't yet handle production scale, and we'll flip the pref via Nimbus), we will try to fetch an ad (that is, a related recommended product) for each product they analyze, to see if the ads server has inventory, but we won't do anything with the ad. If an ad was returned, we will record a Glean event.

- The key question this aims to answer is: how often could users see ads, if they were enabled?

- Data Science will use the ad frequency information to calculate the necessary population sizing for a future experiment where some shopping users will be shown ads.

- Note that these users have already opted-in to the shopping experience, which is a necessary step before we can send URL-equivalent product IDs (the product IDs are unique parts of product page URLs) to the review checker fakespot.com analysis server. Here, we will also be sending the URL-equivalent product IDs to the fakespot ads server. This is consistent with our opt-in vs opt-out data policy.

2) Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements? Some example responses:

- The exposure data recorded in this patch will enable us to accurately size a future experiment that will show ads (that is, recommended related products) in the bottom of the shopping sidebar. Our aim is to make these ads useful, and running an experiment with large enough population to determine statistically significant impacts on usage and retention will tell us if users shown ads tend to stop using the review checker or even stop using firefox. This data will be used to improve the review checker experience and decide how and when to include ads in the final experience.

3) What alternative methods did you consider to answer these questions? Why were they not sufficient?
- Because the ads server inventory is small, it's very difficult to estimate the population sizing for the future ads experiment. This real-world data will tell us how often users can expect to see ads when ads are enabled.

4) Can current instrumentation answer these questions?
- No

5) List all proposed measurements and indicate the category of data collection for each measurement, using the [Firefox data collection categories](https://wiki.mozilla.org/Data_Collection) found on the Mozilla wiki.   

**Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.**

<table>
  <tr>
    <td>Measurement Description</td>
    <td>Data Collection Category</td>
    <td>Tracking Bug #</td>
  </tr>
  <tr>
    <td>On a supported product page, the review checker showed analysis, and the ads exposure pref was enabled, or review checker ads were enabled, and when we tried to fetch an ad from the ad server, an ad was available. Does not indicate whether the ad was actually shown.</td>
    <td>Interaction</td>
    <td>#17306</td>
  </tr>
</table>

6) Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way.
https://dictionary.telemetry.mozilla.org/apps/firefox_ios

7) How long will this data be collected?  Choose one of the following:
- Currently set to expire 2024-06-01

8) What populations will you measure?
- All channels, all locales, all countries

9) If this data collection is default on, what is the opt-out mechanism for users?
- The user can opt out

10) Please provide a general description of how you will analyze this data.
- Glean

11) Where do you intend to share the results of your analysis?
- Glean and ad partners

12) Is there a third-party tool (i.e. not Glean or Telemetry) that you are proposing to use for this data collection? If so:
- No third party tools

**Comment by PARAIPAN9** [comment]
@perrymcmanis144 && @dnarcese. Sorry, I forgot to add the data review form before merging.

**Pull request merged**